validate.go

```diff
@@ -80,17 +83,21 @@ var bundleValidateCommand = cli.Command{
 		}
 
 		hooksCheck := context.Bool("hooks")
-		bundleValidate(spec, rootfsPath, hooksCheck)
+		hostCheck := context.Bool("host-specific")
+		bundleValidate(spec, rootfsPath, hooksCheck, hostCheck)
 		logrus.Infof("Bundle validation succeeded.")
 		return nil
 	},
 }
 
-func bundleValidate(spec rspec.Spec, rootfs string, hooksCheck bool) {
+func bundleValidate(spec rspec.Spec, rootfs string, hooksCheck bool, hostCheck bool) {
 	checkMandatoryField(spec)
 	checkSemVer(spec.Version)
 	checkPlatform(spec.Platform)
```

Comment:
On Mon, Jun 27, 2016 at 12:00:12AM -0700, Ma Shimiao wrote:
> >  	checkPlatform(spec.Platform)
> 
> I think Paltform check has nothing to do with hostCheck

The spec has [1]:

  The runtime MUST generate an error if it does not support the
  configured os.

I guess a VM-based runtime could support launching a Windows bundle
from a Linux host (etc.), but your /proc/filesystems approach here is
assuming runtimes are using a container approach, and not a VM
approach.  With a container approach, a runtime based on the host's
Linux kernel is not going to be able to support a Windows container.

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc1/config.md#platform